src/Unit-3/lesson3/README.md

```diff
@@ -0,0 +1,167 @@
+
+````csharp
+using System;
+using Raven.Client;
+using Raven.Client.Document;
+using Raven.Client.Indexes;
+
+namespace BatchOperationsWithRavenDB
+{
+    public static class DocumentStoreHolder
+        {
+            private static readonly Lazy<IDocumentStore> LazyStore =
+                new Lazy<IDocumentStore>(() =>
+                {
+                    var store = new DocumentStore
+                    {
+                        Url = "http://localhost:8080",
+                        DefaultDatabase = "Northwind"
+                    };
+
+                    store.Initialize();
+
+                    var asm = Assembly.GetExecutingAssembly();
+                    IndexCreation.CreateIndexes(asm, store);
+
+                    return store;
+                });
+
+            public static IDocumentStore Store =>
+                LazyStore.Value;
+        }
+}
+````
+
+We are performing an Assembly scan for indexes. You learned about it in the 
+Unit-2.
+
+### Step 3: Write the model class
+
+As usual, in this excersise, we will write a model class just with the properties 
+we need to perform the batch operation.
+
+It's important to mention that in a "real" world application you  probably would  
+have all the model classes written. Right? 
+
+````csharp
+public class Product
+{
+    public int PricePerUnit { get; set; }
+    public bool Discontinued { get; set; }
+}
+````
+
+### Step 4: Writing the index creation task
+
+You will need to perform a query. As you know, RavenDB uses indexes for all
+queries. So, before querying, we will need to define an index.
+
+````csharp
+public class Products_ByDiscontinued : AbstractIndexCreationTask<Product>
+{
+    public Products_ByDiscontinued()
+    {
+        Map = (products) =>
+            from p in products
+            select new {p.Discontinued};
+    }
+}
+````
+
+This is just a pretty regular index. Nothing special! Note that in a "real world" application
+you would probably have already defined an index that you could use instead 
+of creating a new one.
+
+### Step 5: Performing a batch operation
+
+We now have everything we need to perform an batch operation. Let's do it
+
+````csharp 
+class Program
+{
+    static void Main()
+    {
+        var commands = DocumentStoreHolder.Store.DatabaseCommands;
+        var operation = commands.UpdateByIndex(
+            "Products/ByDiscontinued",
+            new IndexQuery { Query = "Discontinued:false" },
+            new ScriptedPatchRequest
+            {
+                Script = @"this.PricePerUnit = this.PricePerUnit * 1.1"
+            }
+            );
+
+        operation.WaitForCompletion();
+        Console.WriteLine("All active products had Price per unit increased in 10%");
+    }
+}
+````
+
+Please note that we are using a low-level command here. There is no need to have 
+a session.
+
+Run it! And check the results. Amazing!
```

Comment:
Unless index is stale (by default BulkOperationOptions.AllowStale = false). That's fine to assume that in boot camp scenario, however in real world case people can become disappointed. From the other side this is not a good idea to put all of the details here, so I'm not sure what would be the best option (sorry :))